app/models/ElasticRetriever.scala

```
@@ -183,6 +183,68 @@ class ElasticRetriever @Inject()(client: ElasticClient,
         (mappedHits, aggs)
     }
 
+  def getQ[A](esIndex: String,
+              boolQ: BoolQuery,
+              pageSize: Int,
+              buildF: JsValue => Option[A],
+              aggs: Iterable[AbstractAggregation] = Iterable.empty,
+              sortByFields: List[sort.FieldSort] = Nil,
+              excludedFields: Seq[String] = Seq.empty,
+              searchAfter: Option[String] = None): Future[(IndexedSeq[A], Long, Option[String])] = {
+
+    val sa = decodeSearchAfter(searchAfter)
+    val q = search(esIndex)
+      .bool(boolQ)
+      .size(pageSize)
+      .aggs(aggs)
+      .trackTotalHits(true)
+      .sourceExclude(excludedFields)
+      .searchAfter(sa)
+
+    // just log and execute the query
+    val elems: Future[Response[SearchResponse]] = client.execute {
+      val qq = sortByFields match {
+        case Nil => q
+        case _ =>
+          q.sortBy(sortByFields:_*)
+
+      }
+
+      logger.debug(client.show(qq))
+      qq
+    }
+
+    elems.map {
+      case _: RequestFailure                       => (IndexedSeq.empty, 0, None)
```

Failing silently: we should log something here so we can figure out what's happening.

---

Good catch